mixins/d2l-sequences-completion-tracking-mixin.js

```
@@ -18,36 +18,51 @@ function CompletionTrackingMixin() {
 				_completionEntity: {
 					type: Object
 				},
-				_failedCompletion: {
-					type: Object
-				},
 				_skipCompletion: {
 					type: Boolean,
 					computed: '_isImpersonating(token)'
 				}
 			};
 		}
 
-		finishCompletion() {
-			this._fireToastEvent();
-
-			if (!this._completionEntity || this._skipCompletion) {
+		startCompletion() {
+			if (!this.entity || this._skipCompletion || this._isComplete(this.entity)) {
 				return;
 			}
 
-			this._performViewActions(this._completionEntity, 'finish-view-activity')
-				.then(() => { this._completionEntity = null; })
-				.catch(() => { this._completionEntity = null; });
+			this._performViewActions(this.entity, 'view-activity-duration')
+				.then(completion => {
+					this._completionEntity = completion;
+					this._finishCompletion();
+				})
+				.catch(() => {});
 		}
 
-		startCompletion() {
-			if (this._skipCompletion) {
+		startPreviousEntityCompletion(previousEntity) {
+			if (!previousEntity || this._skipCompletion  || this._isComplete(previousEntity)) {
 				return;
 			}
+			// need timeout as items such as quizes, assignments, discussions etc
+			// will not appear complete if immediately fetching activity completion information
+			// after navigation to a new item
+			setTimeout(() => {
+				this._performViewActions(previousEntity, 'view-activity-duration')
+					.then(completion => {
+						this._completionEntity = completion;
+						this._finishCompletion();
+					})
+					.catch(() => {});
```

Same comment about silencing errors.